File: app/src/main/java/io/euphoria/xkcd/app/impl/ui/data/DisplayListener.java
package io.euphoria.xkcd.app.impl.ui.data;

/** Created by Xyzzy on 2020-03-29. */

public interface DisplayListener {

    void notifyItemRangeInserted(int start, int length);

    void notifyItemChanged(int index);

    void notifyItemMoved(int from, int to);

    void notifyItemRangeRemoved(int start, int length);

}

File: app/src/main/java/io/euphoria/xkcd/app/impl/ui/data/DisplayListener.kt
package io.euphoria.xkcd.app.impl.ui.data

/** Created by Xyzzy on 2020-03-29.  */
interface DisplayListener {
    fun notifyItemRangeInserted(start: Int, length: Int)
    fun notifyItemChanged(index: Int)
    fun notifyItemMoved(from: Int, to: Int)
    fun notifyItemRangeRemoved(start: Int, length: Int)
}
